### DISCUSSION TITLE
Is the doc for deployment of WASM to IIS updated ?

### DISCUSSION THREAD

**Discussion by digger73** [unoplatform/uno#13005] [Q&A] [answered]
Hi 

I'm in the process of trying to deploy a simple WASM to an IIS server. 
When trying to add the MIME types it collides with what currently is entered in the web.config. This is the page I'm using [IIS config](https://platform.uno/docs/articles/how-to-host-a-webassembly-app.html?q=server#iis)

The web.config contains the following:
```
    <staticContent>
      <remove fileExtension=".dll" />
      <remove fileExtension=".wasm" />
      <remove fileExtension=".woff" />
      <remove fileExtension=".woff2" />
      <mimeMap fileExtension=".wasm" mimeType="application/wasm" />
      <mimeMap fileExtension=".clr" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".pdb" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".woff" mimeType="application/font-woff" />
      <mimeMap fileExtension=".woff2" mimeType="application/font-woff" />
      <mimeMap fileExtension=".dat" mimeType="application/octet-stream" />
      <!-- Required for PWAs -->
      <mimeMap fileExtension=".json" mimeType="application/octet-stream" />
    </staticContent>

```
Resulting in errors like
![image](https://github.com/unoplatform/uno/assets/81563866/e93c2de9-6ff1-4427-b4c7-ad2776ac8ab3)

Furthermore, it seems that it is necessary to add `<remove fileExtension=".json" />”` otherwise IIS will complain as well

**Comment by jeromelaban** [comment]
Thanks for the report. Depending on the configuration of your current IIS instance, you will need to adjust the list of mime types in the `web.config`. Note that the `.clr` extension is required for the app to start properly, whether it is defined locally for the app, or globally through another configuration file.

**Reply by digger73** [reply]
Thank you for the swift reply.
It is a completely new IIS instance on 2022 server, only added thing is the URL rewriter module, that is described in the doc.
Now you are mentioning locally and globally, I'm not an IIS expert just trying to get it to run ;)
Does it mean that either it should be added as a mime type in the IIS or it should be in the web.config file ? Right now it isn't possible to add to the IIS due to it excising in the web.config. However, I can remove it from the web.config and then added to mime type?

As is now with just using the web.config I only get the splash screen when visiting the app.

**Reply by digger73** [reply, marked as answer]
I solved the problem with the web.config all the mime types the IIS complained about I added `<remove fileExtension=".json" />` just before the definition in the web.config file. this was necessary for: wasm, clr, pdb, dat and json. Now they all got imported into the mime types in the IIS, so I guess the step mentioned regarding adding the mime types isn't necesary as this is covered by the web.config, it just have to get updated.
So the section ended up looking like this:
```

    <staticContent>
      <remove fileExtension=".dll" />    
      <remove fileExtension=".woff" />
      <remove fileExtension=".woff2" />
	<remove fileExtension=".wasm" />
      <mimeMap fileExtension=".wasm" mimeType="application/wasm" />
	<remove fileExtension=".clr" />
      <mimeMap fileExtension=".clr" mimeType="application/octet-stream" />
	<remove fileExtension=".pdb" />
      <mimeMap fileExtension=".pdb" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".woff" mimeType="application/font-woff" />
      <mimeMap fileExtension=".woff2" mimeType="application/font-woff" />
	<remove fileExtension=".dat" />
      <mimeMap fileExtension=".dat" mimeType="application/octet-stream" />
      <!-- Required for PWAs -->
	<remove fileExtension=".json" />
      <mimeMap fileExtension=".json" mimeType="application/octet-stream" />
    </staticContent>
```
